python/ray/ml/trainer.py

```diff
@@ -157,6 +162,51 @@ def __new__(cls, *args, **kwargs):
         trainer._param_dict = {**arg_dict, **kwargs}
         return trainer
 
+    def _validate_attributes(self):
+        """Called on __init()__ to validate trainer attributes."""
+        # Run config
+        if not isinstance(self.run_config, RunConfig):
+            raise TrainerConfigError(
+                f"`run_config` should be an instance of `ray.ml.RunConfig`, "
+                f"found {type(self.run_config)} with value `{self.run_config}`."
+            )
+        # Scaling config
+        # Todo: move to ray.ml.ScalingConfig
+        if not isinstance(self.scaling_config, dict):
+            raise TrainerConfigError(
+                f"`scaling_config` should be an instance of `dict`, "
+                f"found {type(self.run_config)} with value `{self.run_config}`."
+            )
+        # Datasets
+        if not isinstance(self.datasets, dict):
+            raise TrainerConfigError(
+                f"`datasets` should be a dict mapping from a string to "
+                f"`ray.data.Dataset` objects, "
+                f"found {type(self.run_config)} with value `{self.run_config}`."
+            )
+        elif any(not isinstance(ds, ray.data.Dataset) for ds in self.datasets.values()):
```

Comment:
Can we move the `ray.data` import to inside here, and then only do the check if `self.datasets` is not empty?

I'm not sure if we want to require a ray.data dependency in all cases- even when users want to use the Trainer without Ray Datasets.

Reply:
Datasets defaults to `{}` if None was passed, so not using datasets is still an option

Reply:
I'll move to `import ray` instead